src/rest/routes/deferred/deferred-installation-url.ts

```diff
@@ -0,0 +1,40 @@
+import { Request, Response } from "express";
+import { errorWrapper } from "../../helper";
+import { GetDeferredInstallationUrl } from "rest-interfaces";
+import { BaseLocals } from "..";
+import {
+	registerSubscriptionDeferredInstallPayloadRequest,
+	SubscriptionDeferredInstallPayload
+} from "services/subscription-deferred-install-service";
+import { envVars } from "config/env";
+import { InvalidArgumentError } from "config/errors";
+
+export const DeferredInstallationUrlRoute = errorWrapper("GetDeferredInstallationUrl", async function DeferredInstallationUrlRoute(req: Request, res: Response<GetDeferredInstallationUrl, BaseLocals>) {
+	const { gitHubInstallationId, gitHubOrgName } = req.query;
+	const { installation } = res.locals;
+
+	if (!gitHubInstallationId) {
+		req.log.warn("Missing gitHubInstallationId in query");
+		throw new InvalidArgumentError("Missing gitHubInstallationId in query");
+	}
+
+	if (!gitHubOrgName) {
+		req.log.warn("Missing gitHubOrgName in query");
+		throw new InvalidArgumentError("Missing gitHubOrgName in query");
+	}
+
+	const payload: SubscriptionDeferredInstallPayload = {
+		installationIdPk: installation.id,
+		jiraHost: installation.jiraHost,
+		gitHubInstallationId: parseInt(gitHubInstallationId.toString()),
+		orgName: gitHubOrgName.toString(),
+		gitHubServerAppIdPk: undefined // TODO: This only works for cloud, Add this value for GHE servers
+	};
+	const requestId = await registerSubscriptionDeferredInstallPayloadRequest(payload);
```

Comment:
Is it worth wrapping up the payload gen and the async call in a try/catch?

Reply:
We can, but don't think that'll be necessary. 
The only time that this would fail is if redis is not running. And redis is running in docker, so we would already know if it fails from the docker. 
Plus, I haven't seen any try catch being used in the other places where this method is being used.